### ISSUE TITLE
Intensive CPU usage.

### ISSUE THREAD

**Issue by cheshir** [paulbellamy/ratecounter#9]
Hi guys.

After upgrading library from #7 version I caught strange CPU behavior:

![example](http://i.piccy.info/i9/32e88489429e5dd3e0f20059ec7a5062/1500292921/62200/1097549/CPU_usage.jpg)
I couldn't reproduce this bug with small snippet. So keep in mind this issue when you will have the similar problem.

**Comment by paulbellamy** [comment]
Thanks, do you know what rate the code was being called at? regularly? irregular? high-volume? low-volume?

Also, what were the interval and resolution of your counter set to?

**Comment by cheshir** [comment]
In most cases I have a one minute timeout and I'm not using `resolution`.

`Rate` method is called every ~40 seconds.
`Incr` method is called from 0 to ~1000 times per second on each counter.

**Comment by t0mk** [comment]
I might have the same problem. With ratecounter after f3fe9d300be8ba75181604e7da19096fbccfdb2b, after ~1 day of ~10/s Incr()s on ~100 histograms, the process cpu usage rises, and pprof shows that most of the time is spent in runtime.siftdownTimer. I suspect similar issue as in
https://forum.golangbridge.org/t/runtime-siftdowntimer-consuming-60-of-the-cpu/3773

coming from https://github.com/paulbellamy/ratecounter/blob/master/ratecounter.go#L48 

Maybe just rewrite the for loop as
```
ticker := time.NewTicker(time.Duration(float64(r.interval) / float64(r.resolution)))
select { 
case <- ticker.C: 
    [...]
}
```

I will test this and maybe write later.

**Comment by cheshir** [comment]
I've created the pull request with fix #10. @t0mk please check if you have the same problem now.
